HIRS_Utils/src/test/java/hirs/ima/matching/ImaAcceptablePathAndHashRecordMatcherTest.java

```diff
@@ -99,16 +100,65 @@ public void testContainsEquivalentPaths() {
         IMAMeasurementRecord measurementRecord =
                 new IMAMeasurementRecord(LIB64_LD_FILE, USR_LIB64_LD_HASH);
         Assert.assertEquals(
-                new ImaAcceptableRecordMatcher(
+                new ImaAcceptablePathAndHashRecordMatcher(
                         Collections.singletonList(baselineRecord),
                         SimpleImaBaselineTest.getTestImaPolicy(false),
-                        baseline).contains(measurementRecord),
+                        baseline
+                ).contains(measurementRecord),
                 new IMAMatchStatus<>(
                         measurementRecord, ReportMatchStatus.MATCH, baselineRecord, baseline
                 )
         );
     }
 
+    /**
+     * Tests that contains correctly matches equivalent paths.
+     */
+    @Test
+    public void testExhaustiveContainsEquivalentPaths() {
+        SimpleImaBaseline baseline = getTestSimpleImaBaseline();
+
+        List<IMABaselineRecord> baselineRecords = Arrays.asList(
+                new IMABaselineRecord("/usr/sbin/dhclient", ONES),
+                new IMABaselineRecord("/usr/sbin/sysctl", ONES),
+                new IMABaselineRecord("/usr/sbin/swapon", ONES),
+                new IMABaselineRecord("/sbin/audispd", ONES),
+                new IMABaselineRecord("/usr/sbin/sysctl", ONES),
+                new IMABaselineRecord("/sbin/ldconfig", ONES),
+                new IMABaselineRecord("/sbin/kexec", ONES),
+                new IMABaselineRecord("/usr/sbin/ip", ONES),
+                new IMABaselineRecord("/usr/bin/bash", ONES)
+        );
+
+        List<IMAMeasurementRecord> measurementRecords = Arrays.asList(
+                new IMAMeasurementRecord("/sbin/dhclient", ONES),
+                new IMAMeasurementRecord("/sbin/sysctl", ONES),
+                new IMAMeasurementRecord("/sbin/swapon", ONES),
+                new IMAMeasurementRecord("/usr/sbin/audispd", ONES),
+                new IMAMeasurementRecord("/sbin/sysctl", ONES),
+                new IMAMeasurementRecord("/usr/sbin/ldconfig", ONES),
+                new IMAMeasurementRecord("/usr/sbin/kexec", ONES),
+                new IMAMeasurementRecord("/sbin/ip", ONES),
+                new IMAMeasurementRecord("/bin/bash", ONES)
```

Comment:
This tests the equivalence of `/sbin` to `/usr/sbin` and the reverse several times. It also tests the equivalence of `/usr/bin` to `/bin`, but not its reverse. Of the 10 directional equivalences, only 3 are being tested. There are actually more than 10 that should be tested since `/bin` is equivalent to `/usr/bin` and `/usr/bin` is equivalent to `/usr/sbin`, which is equivalent to `/sbin`.

Reply:
Thanks!  I based this unit test off some real-world examples we saw, but I will expand the cases to handle all 10 current directional equivalences.  At this time, we are not considering these equivalencies as transitive, so there are indeed 10 cases to check.

Reply:
Very good.